internal/datastore/test_tables.go

```diff
@@ -0,0 +1,170 @@
+package datastore
+
+import (
+	"context"
+
+	"github.com/nedrocks/delphisbe/graph/model"
+
+	"github.com/sirupsen/logrus"
+)
+
+type TestData struct {
+	Discussions  []model.Discussion
+	Participants []model.Participant
+	Posts        []model.Post
+}
+
+func (d *delphisDB) CreateTestTables(ctx context.Context, data TestData) (func() error, error) {
+	if err := d.createTestTables(ctx); err != nil {
+		logrus.WithError(err).Error("failed to create test tables")
+		return nil, nil
+	}
+
+	if err := d.populateTestTables(ctx, data); err != nil {
+		logrus.WithError(err).Error("failed to populate test tables")
+		return nil, err
+	}
+
+	return d.close, nil
+}
+
+func (d *delphisDB) createTestTables(ctx context.Context) error {
+	d.readyMu.Lock()
+	defer d.readyMu.Unlock()
+
+	createTableQueries := []string{
```

Comment:
Maybe I'm ignorant here but why not use the migrations in _scripts/migrations? It feels like this way the test db will diverge from the true DB schema over time.